### PR TITLE
Add manual testing for undo in DG

### LINKED ISSUE #282: [PE-D][Tester A] Add command taking in less params than Edit command
Users are unable to initialize contacts with the following fields:
1. en/EMERGENCY_NAME
2. g/GRADUATION_YEAR
3. ep/EMERGENCY_PHONE

Users have to first initialise contacts with "add", then use "edit" to add these additional fields, which feels a little bit like a waste, given that "edit" already implements the modification of these fields.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: Domokunx/ped#5

### LINKED ISSUE #305: Add manual testing for undo in DG

### PR DESCRIPTION
**Pull request by charlesliyifeng**
Fixes #305, fixes #282

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -642,6 +642,19 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
+### Undo a command
+
+1. Undoing a `delete` command
+
+   1. Prerequisites: The previous command is a `delete` command
```
Comment (chashaobao123) [inline docs/DeveloperGuide.md:649]: can we add fullstop at the end of the sentence such that it is consistent with the rest?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -642,6 +642,19 @@ testers are expected to do more *exploratory* testing.
...
+   2. Test case: `undo`<br>
+      Expected: The previous deletion is reverted. Details of the restored contact shown in the status message.
+
+1. No command to undo
+
+   1. Prerequisites: No undoable commands have been executed since the starting of the app
+   2. Test case: `undo`<br>
+      Expected: No command is undone. Error message displayed in the status message.
```
Comment (SeanFoongjt) [inline docs/DeveloperGuide.md:657]: Should there be a bit more information on the error message?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -650,3 +663,8 @@ Team size: 5
 1. **Add more precise functionality to the `clean` command**. The `clean` command currently does not allow removal of 
 students who have graduated in the current year, as it can only detect the graduation year but not the month. We plan to 
 add support for storing a more specific graduation date, such that we can accurately remove students who have graduated immediately after their graduation.
+
+1. **Add the support for setting EmergencyName, EmergencyPhone and GraduationYear using the `add` command.**
```
Comment (SeanFoongjt) [inline docs/DeveloperGuide.md:667]: I think it should be Add support and not Add the support here.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -642,6 +642,19 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
+### Undo a command
+
+1. Undoing a `delete` command
```
Comment (SeanFoongjt) [inline docs/DeveloperGuide.md:647]: Should there be one or two more examples of other commands that can be deleted?

**Review by EdaYeo — APPROVED** [review]
LGTM!

**Pull request merged**